mono/utils/mono-threads-state-machine.c

```diff
@@ -134,17 +134,19 @@ mono_threads_transition_detach (MonoThreadInfo *info)
 	switch (cur_state) {
 	case STATE_RUNNING:
 		g_assert (suspend_count == 0);
+		/* fallthrough */
+	case STATE_BLOCKING: /* An OS thread on coop goes STARTING->BLOCKING->RUNNING->BLOCKING->DETACHED */
```

Comment:
this should also check for `g_assert (suspend_count == 0);` as `mono_threads_transition_detach` is called while holding the suspend lock, so the suspend count should be 0.